scripts/pylib/twister/harness.py

```diff
@@ -132,31 +132,14 @@ def handle(self, line):
             self.tests[name] = match.group(1)
             self.ztest = True
 
-        if self.RUN_PASSED in line:
-            if self.fault:
-                self.state = "failed"
-            else:
-                self.state = "passed"
-
-        if self.RUN_FAILED in line:
-            self.state = "failed"
-
-        if self.fail_on_fault:
-            if self.FAULT in line:
-                self.fault = True
+        self.process_test(line)
```

Comment:
I see 8caa7d2e567607c65a542ef66e45ce4bba98c773 move this to line 142, and now it is moved back